tests/jballerina-debugger-integration-test/src/test/java/org/ballerinalang/debugger/test/adapter/variables/VariableQueryTest.java

```diff
@@ -0,0 +1,193 @@
+        //         debugTestRunner.fetchChildVariables(tableChildVariables.get("[0]"));
+        // debugTestRunner.assertVariable(tableGrandChildrenVariables, "id", "1", "int");
+        // tableGrandChildrenVariables =
+        //         debugTestRunner.fetchChildVariables(tableChildVariables.get("[999]"));
+        // debugTestRunner.assertVariable(tableGrandChildrenVariables, "id", "1000", "int");
+
+        // table child variable query test with start = 600 and count = 100
+        tableChildVariables = debugTestRunner.fetchChildVariables(localVariables.get("tableVar"), 600, 100);
+        Assert.assertEquals(tableChildVariables.size(), 100);
+        tableGrandChildrenVariables =
+                debugTestRunner.fetchChildVariables(tableChildVariables.get("[600]"));
+        debugTestRunner.assertVariable(tableGrandChildrenVariables, "id", "601", "int");
+        tableGrandChildrenVariables =
+                debugTestRunner.fetchChildVariables(tableChildVariables.get("[699]"));
+        debugTestRunner.assertVariable(tableGrandChildrenVariables, "id", "700", "int");
+    }
+
+    @Test
+    public void tupleVariableQueryTest() throws BallerinaTestException {
+        // tuple child variable query test
+        Map<String, Variable> tupleChildVariables = debugTestRunner.fetchChildVariables(localVariables.get("tupleVar"));
+        Assert.assertEquals(tupleChildVariables.size(), 1000);
+        debugTestRunner.assertVariable(tupleChildVariables, "[0]", "1", "int");
+        debugTestRunner.assertVariable(tupleChildVariables, "[999]", "1000", "int");
+
+        // tuple child variable query test with start = 0 and count = 0
+        tupleChildVariables = debugTestRunner.fetchChildVariables(localVariables.get("tupleVar"), 0, 0);
+        Assert.assertEquals(tupleChildVariables.size(), 1000);
+        debugTestRunner.assertVariable(tupleChildVariables, "[0]", "1", "int");
+        debugTestRunner.assertVariable(tupleChildVariables, "[999]", "1000", "int");
+
+        // tuple child variable query test with start = 700 and count = 100
+        tupleChildVariables = debugTestRunner.fetchChildVariables(localVariables.get("tupleVar"), 700, 100);
+        Assert.assertEquals(tupleChildVariables.size(), 100);
+        debugTestRunner.assertVariable(tupleChildVariables, "[700]", "701", "int");
+        debugTestRunner.assertVariable(tupleChildVariables, "[799]", "800", "int");
+    }
+
+    @Test
+    public void xmlVariableQueryTest() throws BallerinaTestException {
+        // xml child variable query test
+        Map<String, Variable> xmlChildVariables = debugTestRunner.fetchChildVariables(localVariables.get("xmlVar"));
+        Assert.assertEquals(xmlChildVariables.size(), 1000);
+
+        // xml child variable query test with start = 0 and count = 0
+        xmlChildVariables = debugTestRunner.fetchChildVariables(localVariables.get("xmlVar"), 0, 0);
+        Assert.assertEquals(xmlChildVariables.size(), 1000);
+
+    }
```

Comment:
Don't we need to check intermediate ranges for xml, just like we have done for other types?

Reply:
Fixed with https://github.com/ballerina-platform/ballerina-lang/pull/32015/commits/b385f1235a8159e68de0c9353d344ad184ccd6a4